src/storage/v2/edge_accessor.cpp

```diff
@@ -21,6 +21,33 @@
 namespace memgraph::storage {
 
 bool EdgeAccessor::IsVisible(const View view) const {
+  bool visible = true;
+  // When edges don't have properties, their isolation level is still dictated by MVCC ->
+  // iterate over the deltas of the from_vertex_ and see which deltas can be applied on edges.
+  if (!config_.properties_on_edges) {
+    Delta *delta = nullptr;
+    {
+      std::lock_guard<utils::SpinLock> guard(from_vertex_->lock);
+      visible = !from_vertex_->deleted;
+      delta = from_vertex_->delta;
+    }
+    ApplyDeltasForRead(transaction_, delta, view, [&](const Delta &delta) {
```

Comment:
I like this approach, but we should extend this.
1. We shouldn't just use a single bool variable, because there is a difference between whether the object doesn't exists for the transaction because it is not yet created or it is already deleted
2. The gid of the edge is important, otherwise we might consider deltas that are corresponding to other edges. Check how `delta.vertex_edge.edge` is used.